File: src/main/java/io/openfuture/openmessanger/service/dto/AuthenticatedChallengeRequest.java
package io.openfuture.openmessanger.service.dto;

import javax.validation.constraints.NotBlank;

import org.springframework.lang.NonNull;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AuthenticatedChallengeRequest {

    @NotBlank
    private String sessionId;

    @NonNull
    @NotBlank(message = "username is mandatory")
    private String username;

    @NotBlank
    private String challengeType;
}

File: src/main/kotlin/io/openfuture/openmessanger/service/dto/AuthenticatedChallengeRequest.kt
package io.openfuture.openmessanger.service.dto

import javax.validation.constraints.NotBlank

data class AuthenticatedChallengeRequest(
    private val sessionId: @NotBlank String? = null,
    private val username: @NotBlank(message = "username is mandatory") String? = null,
    private val challengeType: @NotBlank String? = null)
